File: _book/source/_posts/Algorithm/1.leetcode/数据结构/栈/155.最小栈.js
/*
 * @lc app=leetcode.cn id=155 lang=javascript
 *
 * [155] 最小栈
 */

// @lc code=start
/**
 * initialize your data structure here.
 */
var MinStack = function () {
    this.stack = []
    this.minV = Number.MAX_VALUE

};

/** 
 * @param {number} val
 * @return {void}
 */
MinStack.prototype.push = function (val) {
    const minV = this.minV
    if (x < this.minV) {
        this.minV = x
    }
};

/**
 * @return {void}
 */
MinStack.prototype.pop = function () {

};

/**
 * @return {number}
 */
MinStack.prototype.top = function () {

};

/**
 * @return {number}
 */
MinStack.prototype.getMin = function () {

};

/**
 * Your MinStack object will be instantiated and called as such:
 * var obj = new MinStack()
 * obj.push(val)
 * obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.getMin()
 */
// @lc code=end

File: source/_posts/Algorithm/经典算法/单调栈/155.最小栈.py
#
# @lc app=leetcode.cn id=155 lang=python3
#
# [155] 最小栈
#

# @lc code=start
class MinStack:

    def __init__(self):
        """
        initialize your data structure here.
        """
        self.minV = float('inf')
        self.stack = []


    def push(self, val: int) -> None:
        self.stack.append(x-self.minV)
        if x< self.minV:
            self.minV = x

    def pop(self) -> None:
        if not self.stack:
            return
        if tmp<0:
            self.minV -= tmp


    def top(self) -> int:


    def getMin(self) -> int:



# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(val)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()
# @lc code=end

